### ISSUE TITLE
ssh authorized keys should have a command restriction possible

### ISSUE THREAD

**Issue by Firstyear** [kanidm/kanidm#197]
It should be possible to store keys with options like :

```
command="borg serve --restrict-to-repository /home/backup/",restrict
```

This way wecan store keys for service accounts centrally and potentially even hav policies on the keys command?

**Comment by ryankurte** [comment]
idk if this is the correct place but, an alternative is to use ssh certificates with associated restrictions, then provide an kani identity agent on the client that swaps credentials for an transient certificate, roles enforced centrally without having to store or distribute keys.

(see also @stoggi's https://github.com/stoggi/sshrimp)

**Comment by Firstyear** [comment]
Yeah, ssh certs is an option as well, but there are different requirements there. I do want ssh certs of course! I think that the ssh certs are a bit more "always online" where the auth-keys gives you some offline capable processing options. But I'll flesh that all out here https://github.com/kanidm/kanidm/issues/198

**Comment by Firstyear** [comment]
Saying that maybe it's better to focus on ssh certs long run so I'll close this ...

**Comment by rdelaage** [comment]
Hello, I had trouble to add a SSH public key with the `permitopen` option. I think this is because of the SSH key verification and this is not implemented yet. Is this an accessible first contribution? If yes can you give me some hints on what to modify?

**Comment by yaleman** [comment]
@rdelaage it'll likely be a bit challenging since we rely on an external crate for parsing/validating them, and that doesn't have any obvious handling for pubkeys with additional options

**Comment by rdelaage** [comment]
I would say that the options are not part of the pubkey itself but should be stored in a dedicated field (maybe an HashMap<String, Vec<String>>) with a predefined list of allowed options
